### PR TITLE
Improve examples handling

### PR DESCRIPTION
**Pull request by ivankravets**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -49,14 +49,16 @@ env:
...
-    - unzip /tmp/xmem2.zip -d /tmp
-    - unzip /tmp/generic_storage.zip -d /tmp
-    - unzip /tmp/RTClib.zip -d /tmp
+
+    # Use PlatformIO from development branch
+    # @TODO remove this line after 2.0.1 release
+    - pip install https://github.com/platformio/platformio/archive/develop.zip
+
+    # Libraries from PlatformIO Library Register
+    # http://platformio.org/#!/lib/show/416/TinyGPS
```
Comment (Lauszus) [inline .travis.yml]: What is this library used for?

Comment (ivankravets) [inline .travis.yml]: This library for ``PLATFORMIO_CI_SRC=examples/pl2303/pl2303_gps/pl2303_gps.ino`` example.

Comment (Lauszus) [inline .travis.yml]: Ahh okay. Forgot about that one ;)

**Comment by Lauszus** [comment]
Hmm this example is still failing: https://travis-ci.org/felis/USB_Host_Shield_2.0/jobs/63768507. It looks like PlatformIO does not include in these files: https://github.com/arduino/Arduino/tree/master/hardware/arduino/sam/cores/arduino/avr?

I can compile it fine in the Arduino IDE after https://github.com/felis/USB_Host_Shield_2.0/commit/44d3b4a3223fc125b01370482b98e6f1ccbba370.

**Comment by ivankravets** [comment]
@Lauszus Sure, because your repo uses obsolete ``.travis.yaml``, please accept my pull request :blush:

**Comment by Lauszus** [comment]
I am linking to the build job for your request ;) I have been pushing some changes to master and the re-tricker the build.

**Comment by Lauszus** [comment]
Anyway. I think I found the issue ;)

**Comment by Lauszus** [comment]
Fixed it: https://github.com/felis/USB_Host_Shield_2.0/commit/5ee8ee93251f02283c2695224bde9b3965a49a73.

**Comment by Lauszus** [comment]
Now they all should pass :) Thanks a lot for your hard work!

**Comment by ivankravets** [comment]
> Now they all should pass :) Thanks a lot for your hard work!

Thank you too! As we can see, PlatformIO helped us to fix 2 examples :smile: 

P.S: I used this project for PlatformIO examples  file:///Volumes/SOFT/Projects/GitHub/platformio/origin/docs/_build/html/ci/travis.html . This is good example to show how to test library examples on the different development platforms (AVR, Teensy, SAM).

**Comment by ivankravets** [comment]
All completed https://travis-ci.org/ivankravets/USB_Host_Shield_2.0/builds/63772053

P.S: If you will have any problems in future, please don't disable Continuous Integration and don't hesitate to contact me. This functionality was released in PlatformIO 2.0 and has not been tested properly. However, I'm ready to polish it and propose for end embedded developers. It should save a lot of time to avoid simple mistakes when team work on the same project.

**Comment by Lauszus** [comment]
> Thank you too! As we can see, PlatformIO helped us to fix 2 examples :smile:
Yes it's pretty sweet. It helps a lot when you support multiple different platform as this project does :)

> P.S: I used this project for PlatformIO examples http://docs.platformio.org/en/latest/ci/travis.html . This is good example to show how to test library examples on the different development platforms (AVR, Teensy, SAM).

Yeah I saw that earlier as well :)

> P.S: If you will have any problems in future, please don't disable Continuous Integration and don't hesitate to contact me. This functionality was released in PlatformIO 2.0 and has not been tested properly. However, I'm ready to polish it and propose for end embedded developers. It should save a lot of time to avoid simple mistakes when team works on the same project.

I won't. This makes a lot of sense for this project, so we don't have to test the code for all the different platforms manually.

**Comment by ivankravets** [comment]
@Lauszus Thanks! :beer: :blush: 

P.S: I see that a few jobs has been cancelled by Travis like "Connection Timeout". Hm... need to contact with @travis-ci and discuss the mirror of PlatformIO packages within Travis infrastructure. It should save a lot of traffic between Travis & SourceForge, and of course, should speed up a build process.

**Comment by Lauszus** [comment]
Yes. I had to restart them manually. Sounds good. Keep me updated :)

**Pull request merged**